Assets/Shaders/MeshPainting/Unwrap.vert

```diff
@@ -5,15 +5,23 @@
 
 #include "../Defines.glsl"
 
+struct SUnwrapData
+{
+	vec4 TargetPosition;
+	vec4 TargetDirection;
+};
+
 layout(push_constant) uniform TransformBuffer
 {
-    uint index;
+	uint index;
 } p_TransformIndex;
 
-layout(binding = 0, set = BUFFER_SET_INDEX) uniform PerFrameBuffer              { SPerFrameBuffer val; }    u_PerFrameBuffer;
+layout(binding = 0, set = BUFFER_SET_INDEX) uniform PerFrameBuffer				{ SPerFrameBuffer val; }	u_PerFrameBuffer;
+
+layout(binding = 0, set = DRAW_SET_INDEX) restrict readonly buffer Vertices		{ SVertex val[]; }			b_Vertices;
+layout(binding = 1, set = DRAW_SET_INDEX) restrict readonly buffer Instances	{ SInstance val[]; }		b_Instances;
 
-layout(binding = 0, set = DRAW_SET_INDEX) restrict readonly buffer Vertices     { SVertex val[]; }          b_Vertices;
-layout(binding = 1, set = DRAW_SET_INDEX) restrict readonly buffer Instances    { SInstance val[]; }        b_Instances;
+layout(binding = 0, set = 3) uniform UnwrapData									{ SUnwrapData val; }		u_UnwrapData;
```

Comment:
I'm not sure what rules you have when writing shaders, but the other set specifiers use macros.

Reply:
Not sure either, but since this set is only used in this shader a macro for it seems a bit unnecessary.